File: app/src/main/java/com/fractalwrench/crazycats/image/list/ImageListView.java
package com.fractalwrench.crazycats.image.list;

import android.support.annotation.Nullable;

import com.fractalwrench.crazycats.image.ContentLoadingView;
import com.fractalwrench.crazycats.image.data.ImageData;

import java.util.List;

/**
 * Displays a scrollable View of Images
 */
public interface ImageListView extends ContentLoadingView<List<ImageData>> {

    void setDelegate(@Nullable CellDelegate delegate);

    void showImageDetail(ImageData imageData);

    interface CellDelegate {
        void onImageCellClicked(ImageData imageData);
    }

}

File: app/src/main/java/com/fractalwrench/crazycats/image/list/ImageListView.kt
package com.fractalwrench.crazycats.image.list

import com.fractalwrench.crazycats.image.ContentLoadingView
import com.fractalwrench.crazycats.image.data.ImageData

/**
 * Displays a scrollable View of Images
 */
interface ImageListView : ContentLoadingView<List<ImageData>> {

    fun setDelegate(delegate: CellDelegate?)

    fun showImageDetail(imageData: ImageData)

    interface CellDelegate {
        fun onImageCellClicked(imageData: ImageData)
    }

}
